packages/core/core-flows/src/product/workflows/create-products.ts

```diff
@@ -4,19 +4,48 @@ import {
   PricingTypes,
   ProductTypes,
 } from "@medusajs/framework/types"
-import { ProductWorkflowEvents, isPresent } from "@medusajs/framework/utils"
+import {
+  ProductWorkflowEvents,
+  isPresent,
+  MedusaError,
+} from "@medusajs/framework/utils"
 import {
   WorkflowData,
   WorkflowResponse,
   createHook,
   createWorkflow,
   transform,
+  createStep,
 } from "@medusajs/framework/workflows-sdk"
 import { emitEventStep } from "../../common"
 import { associateProductsWithSalesChannelsStep } from "../../sales-channel"
 import { createProductsStep } from "../steps/create-products"
 import { createProductVariantsWorkflow } from "./create-product-variants"
 
+interface ValidateProductInputStepInput {
+  products: CreateProductWorkflowInputDTO[]
+}
+
+const validateProductInputStepId = "validate-product-input"
+/**
+ * This step validates a product data before creation.
+ */
+const validateProductInputStep = createStep(
+  validateProductInputStepId,
+  async (data: ValidateProductInputStepInput) => {
+    const { products } = data
+
+    for (const product of products) {
+      if (!product.options?.length) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Product options are not provided for ${product.title}`
+        )
+      }
+    }
+  }
+)
```

Comment:
```suggestion
const missingOptionsProductTitles = products
  .filter((product) => !product.options?.length)
  .map((product) => product.title)

if (missingOptionsProductTitles.length) {
  throw new MedusaError(
    MedusaError.Types.INVALID_DATA,
    `Product options are not provided for [${missingOptionsProductTitles.join(
      ", "
    )}]`
  )
}
```

Reply:
it would prevent having many products with wrong options and only receiving the product error one by one :)